osgp/platform/osgp-domain-core/src/main/java/org/opensmartgridplatform/domain/core/entities/SmartMeter.java

```diff
@@ -14,13 +15,13 @@
 import org.opensmartgridplatform.domain.core.valueobjects.Address;
 import org.opensmartgridplatform.domain.core.valueobjects.GpsCoordinates;
 
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode(callSuper = true)
```

Comment:
I don't think the `EqualsAndHashCode` provided by Lombok is appropriate for entities.
The equals implementation in `Device` takes into account the business key (`deviceIdentification`) which is the correct way to implement equality if such a business key is available.
Lombok adds in the other fields when building its equals implementation. A SmartMeter is not a different SmartMeter if any of these other fields have different values.

Reply:
you are correct. good to know.